.github/workflows/release-build-native-win64.yml

```diff
@@ -0,0 +1,98 @@
+      run: |
+        %JAVA_HOME%/bin/gu.cmd install native-image
+      shell: cmd
+    - name: Configure Pagefile
+      # Increased the page-file size due to memory-consumption of native-image command
+      # For details see https://github.com/actions/virtual-environments/issues/785
+      uses: al-cheb/configure-pagefile-action@v1.2
+    - name: Set version
+      run: mvnw versions:set -DnewVersion="${{ github.event.release.tag_name }}"
+      shell: cmd
+    - name: Build native executable
+      # Invoke the native-image build with the necessary Visual Studio tooling/environment intialized
+      run: |
+        call "C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\Auxiliary\Build\vcvarsall.bat" x86_amd64
+        mvnw package -Dnative"
+      shell: cmd
+    - name: Upload native executable
+      id: upload-native-executable
+      uses: actions/upload-release-asset@v1
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      with:
+        upload_url: ${{ github.event.release.upload_url }}
+        asset_path: target/starfish-${{ github.event.release.tag_name }}-runner.exe
+        asset_name: starfish.exe
+        tag: ${{ github.ref }}-win64
+        asset_content_type: application/octet-stream
+
+# Trigger this job only on Push or Pull-Request
+  build-on-push-pr:
+    if: github.event_name!='release'&&( github.event_name == 'pull_request'|| github.event_name == 'push')
+    name: Build on Push or PR
+    runs-on: windows-2019
+    steps:
+      - uses: actions/checkout@v2
+      - name: Install graalvm
+        uses: DeLaGuardo/setup-graalvm@3
+        with:
+          graalvm-version: '20.0.0.java11'
+      - name: Install native-image
+        run: |
+          %JAVA_HOME%/bin/gu.cmd install native-image
+        shell: cmd
+      - name: Configure Pagefile
+        # Increased the page-file size due to memory-consumption of native-image command
+        # For details see https://github.com/actions/virtual-environments/issues/785
+        uses: al-cheb/configure-pagefile-action@v1.2
+      - name: Set version
+        run: ./mvnw versions:set -DnewVersion="1.0.0" 
+        shell: cmd
+      - name: Build native executable
+        # Invoke the native-image build with the necessary Visual Studio tooling/environment intialized
+        run: |
+          call "C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\Auxiliary\Build\vcvarsall.bat" x86_amd64
+          mvnw package -Dnative"
+        shell: cmd
+      - name: Create Release
+        id: create_release
+        uses: actions/create-release@v1.0.0
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          tag_name: commit-${{ github.sha}}-${{ github.run_id }}
+          release_name: Release commit-${{ github.sha }}-${{ github.run_id }}
+          draft: false
+          prerelease: false
+
+      - name: Upload native executable
+        uses: shogo82148/actions-upload-release-asset@v1
```

Comment:
this reads as if a release is uploaded on a pullrequest - here it should just attach it to the build results.